### PR TITLE
SYN-6078: view worldreadable flag when making views in storm

### PR DESCRIPTION
**Pull request by MichaelSquires**
- Added `worldreadable` arg/flag to cortex, `$lib.view.add`, and `view.add`.
- Added tests

Resolves SYN-6078

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -4400,11 +4400,11 @@ async def _initCoreViews(self):
 
         self._calcViewsByLayer()
 
-    async def addView(self, vdef, nexs=True):
```
[the remark this answers is not in the document]
Comment (MichaelSquires) [reply synapse/cortex.py:4403]: Sure. I'm fine with using it either way.

File: synapse/lib/stormtypes.py [hunk 2]
```diff
@@ -6714,7 +6714,8 @@ class LibView(Lib):
          'type': {'type': 'function', '_funcname': '_methViewAdd',
                   'args': (
                       {'name': 'layers', 'type': 'list', 'desc': 'A list of layer idens which make up the view.', },
-                      {'name': 'name', 'type': 'str', 'desc': 'The name of the view.', 'default': None, }
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the view.', 'default': None, },
+                      {'name': 'worldreadable', 'type': 'boolean', 'desc': 'Grant read access to the `all` role.', 'default': False, },
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:6718]: Do we need to add a `--worldreadable` to the `view.add` command? Or leave it as-is and let people manually opt into this when constructing a view via API?

Comment (MichaelSquires) [reply synapse/lib/stormtypes.py:6718]: Not sure I understand your comment. I did add a `--worldreadable` to `view.add`. Did I miss something somewhere else?

File: synapse/tests/test_cortex.py [hunk 3]
```diff
@@ -6064,12 +6064,19 @@ async def test_norms(self):
 
     async def test_addview(self):
         async with self.getTestCore() as core:
+            allrole = core.auth.allrole
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py:6067]: checking the role directly works; it is a bit unusual to be testing hiverole directly.
would a lowuser [ without explicit privileges ] being checked if they can read the view work too?

File: synapse/lib/stormtypes.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/lib/stormtypes.py:6718]: You're good! i misread :)

File: synapse/tests/test_cortex.py (hunk 3 above)
Comment (MichaelSquires) [reply synapse/tests/test_cortex.py:6067]: yep, that should work. I'll update.

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**